lib/qr_code/matrix.ex

```diff
@@ -191,20 +191,18 @@ defmodule QRCode.Matrix do
   """
   @spec update_map(matrix, list(index), {:ok, element}) :: Result.t(String.t(), matrix)
   def update_map(matrix, positions, submatrix) do
-    {row, col, check_size} = check_size(matrix, submatrix, positions)
-
-    case check_size do
-      true ->
-        Enum.reduce(positions, matrix, fn {row_pos, col_pos}, acc ->
-          and_then2(
-            acc,
-            submatrix,
-            &update(&1, {row_pos, col_pos}, {row_pos + row - 1, col_pos + col - 1}, &2)
-          )
-        end)
-
-      false ->
-        {:error, "Bad value of position {#{row}, #{col}}. Submatrix is out of matrix!"}
+    {row, col, check_size} = and_then2(matrix, submatrix, &check_size(&1, &2, positions))
```

Comment:
Co je divny, ze ve specifikaci je `matrix` definovan jako `[[pos_integer]]`, ale je pouzit ve fci `and_then2`, kde je pouzit jako result a dialyzer neremca. :/

Reply:
uhh, to je. Ja se nad tim celym chci jeste zamyslet a udelat to lepe (jestli to pujde).